### PR TITLE
19 save provenance metadata

### PR DESCRIPTION
**Pull request by gschwend**

### CODE REVIEW

File: rail_scripts/prepare-pz-test [hunk 1]
```diff
@@ -0,0 +1,108 @@
+#!/usr/bin/env python3
```
Comment (crisingulani) [inline rail_scripts/prepare-pz-test:1]: I would leave this script inside the htcondor directory. It is specific for runs with Condor and has dependencies on files that are in `scheduler_examples/htcondor/`.

File: rail_scripts/prepare-pz-test [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+    print(f'Process ID: {args.process_id}')
+    if not args.comment:
+        args.comment = ' --- ' 
+    print(f'User: {args.user}')
+    print(f'Description: {args.comment}')
+    print('---------------------------------------')
+    print()
+    process_dir = f'./{args.process_id}' 
+    os.system(f'mkdir {process_dir}')
+    os.system(f'mkdir {process_dir}/log')
```
Comment (crisingulani) [inline rail_scripts/prepare-pz-test:80]: `os` library has a methods to create dir: `os.makedirs` or `os.mkdir`.

File: rail_scripts/prepare-pz-test [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+    os.system(f'mkdir {process_dir}/input')
+    os.system(f'mkdir {process_dir}/output')
+   
+
+    pz_dir = os.environ['PZPATH']
+    os.system(f'ln -s {pz_dir}/../scheduler_examples/htcondor/rail-condor/rail-condor.py ./{args.process_id}/rail-condor.py')
```
Comment (crisingulani) [inline rail_scripts/prepare-pz-test:86]: `os` library has method to create symbolic link: `os.symlink`

File: scheduler_examples/htcondor/rail-condor/rail-condor.py [hunk 4]
```diff
@@ -85,5 +88,24 @@ def main():
...
                 run([rail_estimate, '--bins=301', '--algorithm=%s' % algorithm,
                     input, output], check=True)
-
+              
+    end_time = datetime.now()
+    duration = end_time - start_time
+    if len(glob.glob('process_info.yaml')) > 0:    
+        # Open the file in append & read mode ('a+')
+        with open('process_info.yaml', "a+") as file_object:
+            file_object.write(f'start_time:{start_time}')
```
Comment (crisingulani) [inline scheduler_examples/htcondor/rail-condor/rail-condor.py:97]: I suggest using the standard way of writing data to yaml.

File: rail_scripts/prepare-pz-test [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+    
+    # WRITE info FILE  
+    yaml_file = f'./{args.process_id}/process_info.yaml'
+    with open(yaml_file, 'w') as file:
+        file.write(yaml.dump(vars(args)))
+    print(f'Process info saved in {yaml_file}')    
+    print() 
+    print() 
+    print('Now, to run the photo-z estimator, excecute: ' )
+    print()
```
Comment (crisingulani) [inline rail_scripts/prepare-pz-test:99]: it would be nice to inform where we should run the command

**Review by crisingulani — CHANGES_REQUESTED** [review]
@gschwend I made some suggestions

File: rail_scripts/prepare-pz-test (hunk 1 above)
Comment (gschwend) [reply rail_scripts/prepare-pz-test:1]: I agree. Also, update setup instructions so it is included in PATH (the reason I put it inside rail_scripts originally).

**Comment by gschwend** [comment]
@crisingulani everything done!

**Review by crisingulani — APPROVED** [review]

**Pull request merged**